values.go

```diff
@@ -42,23 +43,45 @@ type Getter interface {
 // Optional interface to indicate boolean flags that don't accept a value, and
 // implicitly have a --no-<x> negation counterpart.
 type boolFlag interface {
-	Value
 	IsBoolFlag() bool
 }
 
 // Optional interface for arguments that cumulatively consume all remaining
 // input.
 type remainderArg interface {
-	Value
 	IsCumulative() bool
 }
 
 // Optional interface for flags that can be repeated.
 type repeatableFlag interface {
-	Value
 	IsCumulative() bool
 }
 
+// Text is the interface to the dynamic value stored in a flag.
+// (The default value is represented as a string.)
+//
+// If a Text has an IsBoolFlag() bool method returning true, the command-line
+// parser makes --name equivalent to -name=true rather than using the next
+// command-line argument, and adds a --no-name counterpart for negating the
+// flag.
+type Text interface {
+	encoding.TextMarshaler
+	encoding.TextUnmarshaler
+}
+
+type wrapText struct {
+	text Text
+}
+
+func (w wrapText) String() string {
+	buf, _ := w.text.MarshalText()
+	return string(buf)
+}
+
+func (w *wrapText) Set(s string) error {
+	return w.text.UnmarshalText([]byte(s))
+}
+
```

Comment:
You'll need to implement `IsCumulative()` on `wrapText`, and conditionally forward it to `w.text`.

Reply:
oh

Reply:
:thinking: `UnmarshalText` is not implemented to parse one item, but to parse "whole" items.
i.e. `Value.Set` cumulates item, but `UnmarshalText` overwrite all of items.

The wrapText should not implement `IsCumulative`, should it?

Reply:
I want to put `MarshalText` to use, but I have no idea.

Reply:
Sorry, I meant the `IsBoolFlag()` method of the `boolFlag` interface. But I'm not 100% sure that's useful, so maybe don't worry about it.